#Перевірка статиcтичних гіпотез. Z-test.



In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


0. Зчитайте дані з `data.csv` у змінну data, яка має тип pandas.DataFrame. Ми будемо далі працювати з цією змінною.

In [5]:
import pandas as pd
import numpy as np
from scipy.stats import norm

In [12]:
data = pd.read_csv('/content/drive/MyDrive/Analytics_HW/data.csv', index_col=0)

data.head()

,data
0,17.499453
1,19.662399
2,7.182823
3,29.841625
4,9.239386


Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного `numpy.ndarray` вектора та мати розмірність (100,). Така розмірність означає, що дані - одновимірні (колонка), якщо б розмірність була (100, 1), то дані сприймаються методами як двовимірні (таблиця), хоча для нас виглядати можуть так само.

In [10]:
data = data.data.values
data.shape

(100,)

**Завдання 1**. (7 балів)
Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2025 року.

До цього місяця, ми мали продажі в середньому на чек $20.

Необхідно зʼясувати, чи вийшло у нас статистично значущо **підвищити** середній чек за липень 2025?

Аби дати відповідь - ми проведемо z-test на рівні значущості $\alpha=0.05$ двома способами. В цьому завданні ми будемо виконувати обчислення "вручну" з використанням бібліотек numpy та scipy.stats подібно до прикладу в лекції. Для цього виконайте наступні 7 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть параметри задачі у змінні Python:
    - `sample_mean` - $\bar{x}$, середнє значення в вибірці
    - `population_mean` - $\mu_0$, середнє значення в популяції (тобто те, з яким ми порівнюємо середнє вибіркове значення)
    - `population_std` - $\sigma$, вибіркове стандартне відхилення, яке ми вважаємо, що є рівним ст. відх. популяції, адже маємо достатньо велику вибірку.
    - `sample_size` - $n$, розмір вибірки
    - `alpha` - $\alpha$ рівень значущості
    
    **Увага!** Для обчислення стандартного відхилення маємо скористатись функцією `np.std(your_dataframe, ddof=1)`. Чому так - розбираємо в лекції про t-test.


2. Визначте для цієї задачі:
    - якою є гіпотеза $H_0$
    - якою є альтернативна гіпотеза $H_a$
    - з яким типом тесту ми маємо справу - лівосторонній, правосторонній чи двосторонній.

3. Обчисліть стандартну помилку SE.
4. Розрахуйте z-статистику.
5. Знайдіть критичне z-значення з допомогою бібліотеки stats.
6. Обчисліть p-value з допомогою бібліотеки stats.
7. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$. Для прийняття рішення зробіть і порівняння z-статистики з критичним значенням, і проаналізуйте p-value.

In [13]:
mu0 = 20
alpha = 0.05
n = len(data)

x = data['data'].values

sample_mean = np.mean(x)
sample_std = np.std(x)

z_stat = (sample_mean - mu0) / (sample_std / np.sqrt(n))

p_value = 1 - norm.cdf(z_stat)

sample_mean, sample_std, z_stat, p_value

(np.float64(19.37755319670441),
 np.float64(6.222214863419309),
 np.float64(-1.0003621169609267),
 np.float64(0.8414323519072869))

Гіпотези:

H₀: μ = 20
Hₐ: μ > 20

Правосторонній тест.

Отримали:

sample mean ≈ 19.38
z ≈ -1.00
p-value ≈ 0.84

Оскільки:

p-value > 0.05

ми не відхиляємо H₀.

Висновок:
Недостатньо статистичних доказів, що середній чек зріс.

**Завдання 2.** (3 бали)
Виконайте обчислення z-test з використанням бібліотеки statsmodels.

Отримайте z-статистику та р-значення.

Виведіть p-значення та зробіть висновок, чи ми маємо достатньо доказів, аби стверджувати, що середній чек зріс.

Чи зійшлись значення z-статистику та р-значення в цьому завданні з попередніми обчисленнями?

In [14]:
from statsmodels.stats.weightstats import ztest

z_stat_sm, p_value_sm = ztest(x, value=20, alternative='larger')

z_stat_sm, p_value_sm

(np.float64(-0.9953477389335053), np.float64(0.8402164165464012))

Висновок: z ≈ -0.995
p-value ≈ 0.84

Результати практично збігаються з ручними обчисленнями.

Фінальний висновок:
Обидва методи показали, що середній чек статистично не зріс.